frontend/src/redux/history/methods.ts

```diff
@@ -61,11 +57,18 @@ export const deleteAllHistoryEntries = (): Promise<unknown> => {
  * @param newEntry The modified history entry.
  */
 export const updateHistoryEntryRedux = (noteId: string, newEntry: HistoryEntry): void => {
-  store.dispatch({
-    type: HistoryActionType.UPDATE_ENTRY,
+  const entryToUpdate = store.getState().history.find((entry) => entry.identifier === noteId)
+  if (!entryToUpdate) {
+    return
+  }
+  const action = historyActionsCreator.updateEntry({
     noteId,
-    newEntry
-  } as UpdateEntryAction)
+    newEntry: {
+      ...newEntry,
+      origin: entryToUpdate.origin
+    }
+  })
+  store.dispatch(action)
```

Comment:
This shouldn't happen here but in the reducer.

Reply:
Changes